2.6/root/usr/share/container-scripts/mongodb/init-replset.sh

```diff
@@ -69,16 +47,10 @@ function initiate() {
     wait_for_mongo_up ${node} &>/dev/null
   done
 
-
-  if mongo --eval "quit(db.isMaster().setName == '${MONGODB_REPLICA_NAME}' ? 0 : 1)" --quiet; then
-    info "Replica set '${MONGODB_REPLICA_NAME}' already exists, skipping initialization"
-    return
-  fi
-
   local config="{_id: '${MONGODB_REPLICA_NAME}', $(replset_config_members "${current_endpoints}")}"
 
   info "Initiating MongoDB replica using: ${config}"
-  mongo admin --eval "rs.initiate(${config});${WAIT_PRIMARY_STATUS}" --quiet
+  mongo --eval "rs.initiate(${config}); while (!rs.isMaster().ismaster) { sleep(100); }" --quiet
```

Comment:
I'm not convinced that setting an arbitrary timeout is better than having an infinite loop.  In the statefulset case, which I think is the case that people should really be using, the net effect of an infinite loop vs crashing out is the same: the 1st pod never becomes ready, the others don't deploy, the issue is clear from the pod logs.